src/charts/c3/c3.directive.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * @ngdoc directive
+ * @name patternfly.charts.directive:c3Chart
+ *
+ * @description
+ *   Directive for wrapping c3 library
+ *
+ *
+ * @param {string} id the ID iof the container that the chart should bind to
+ * @param {expression} config the c3 configuration options for the chart
+ *
+ * @example
+ <example module="patternfly.charts">
+<file name="index.html">
+  <div ng-controller="ChartCtrl">
+    <div c3-chart id="chartId"  config="chartConfig"></div>
+  </div>
+ </file>
+
+<file name="script.js">
+ function ChartCtrl($scope) {
+    $scope.chartConfig = {"donut":
+                     {"title":"MHz Used",
+                      "label":{"show":false},
+                      "width":10
+                     },
+                     "size":{"height":130},
+                     "legend":{"show":false},
+                     "color":{"pattern":["#CC0000","#D1D1D1"]},
+                     "tooltip":{},
+                     "data":{"columns":[["Used","950"],["Available",50]],
+                     "type":"donut",
+                     "donut":{
+                         "label":{"show":false}
+                      },
+                      "groups":[["used","available"]],
+                      "order":null
+                      }
+                  };
+ }
+ </file>
+ </example>
+ */
+(function(c3){
+  'use strict';
+
+  angular.module('patternfly.charts')
+  .directive('c3Chart', ['$timeout', function($timeout) {
+
+    return {
+      restrict: 'A',
+      scope: {
+        config: '='
+      },
+      template: '<div id=""></div>',
+      replace: true,
+      link: function(scope, element, attrs) {
+        //generate c3 chart data
+        var chartData = scope.config;
+        chartData.bindto = '#' + attrs.id;
+
+        //Generating the chart
+        $timeout(function() {
+          c3.generate(chartData);
+        }, 100);
+      }
```

Comment:
I don't believe this is going to update the chart if the data changes. I think you need to add a watcher on "config" that calls chart.load(new data) if we detect a data change.